Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -135,9 +136,30 @@ The password must meet at least three out of the following four complexity rules
 3. at least one digit (0-9)
 4. at least one special character
 
-Verify the existences of a password policy and password complexity requirements and verify also with the [OWASP Authentication Cheat Sheet](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Password_Complexity "Password Complexity"). Identify all password-related functions in the source code and make sure that a the verification check is performed in each of them. Review the password verification function and make sure that it rejects passwords that violate the password policy.
+Confirm the existence of a password policy and verify the implemented password complexity requirements according to the [OWASP Authentication Cheat Sheet](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Password_Complexity "Password Complexity"). Identify all password-related functions in the source code and make sure that a the verification check is performed in each of them. Review the password verification function and make sure that it rejects passwords that violate the password policy.
+
+[zxcvbn](https://github.com/dropbox/zxcvbn "zxcvbn") is a common library that can be used for estimating password strength, inspired by password crackers. It is available in JavaScript but also for many other programming languages on the server side. Once installed via `bower`, zxcvbn can be used to calculate the complexity and the amount of guesses to crack the password.
```

Comment:
bower is no longer supported right? is it avaialble for yarn?

Reply:
Couldn't find anything for yarn, seems only available via bower. I changed it to a generic sentence and referred to the Github repo as this summarises all the installation methods.

Reply:
Couldn't find anything for yarn, seems only available via bower. I changed it to a generic sentence and referred to the Github repo as this summarises all the installation methods.